docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,444 @@
+---
+
+layout: docs
+
+title:  "Connectable Operators"
+
+section: "chisel3"
+
+---
+
+## Terminology
+
+- "Chisel type" - a `Data` that is not bound to hardware (more details[here](chisel-type-vs-scala-type)).
+  - E.g. `UInt(3.W)`, `new Bundle {..}`, `Vec(3, SInt(2.W))` are all Chisel types
+- `Aggregate` - a Chisel type or component that contains other Chisel types or components (i.e. `Vec`, `Record`, or `Bundle`)
+- `Element` - a Chisel type or component that does not contain other Chisel types or components (e.g. `UInt`, `SInt`, `Clock`, `Bool` etc.)
+- "component" - a `Data` that is bound to hardware (`IO`, `Reg`, `Wire`, etc.)
+  - E.g. `Wire(UInt(3.W))` is a component, whose type is `UInt(3.W)`
+- "member" - a Chisel type or component or any of its children (could be an `Aggregate` or an `Element`)
+  - E.g. `Vec(3, UInt(2.W))(0)` is a member of the parent `Vec` Chisel type
+  - E.g. `Wire(Vec(3, UInt(2.W)))(0)` is a member of the parent `Wire` component
+
+For more details about these Scala types vs Chisel types, please read 
+
+## Overview
+
+The `Connectable` operators are the standard way to connect Chisel hardware components to one another.
+
+> Note: For descriptions of the semantics for the previous operators, see [`Connection Operators`](connection-operators).
+
+All connection operators require the two hardware components (consumer and producer) to be Chisel type-equivalent (matching bundle field names and types (`Record` vs `Vector` vs `Element`), vector sizes, `Element` types (UInt/SInt/Bool/Clock etc)). Use `DataMirror.checkTypeEquivalence` to check this property.
```

Comment:
should this sentence say something about alignments needing to match? This is confusingly ambiguous throughout the article. Does "Chisel type equivalence" include alignment check or not? Later you use a term "structurally equivalent" which you never define here. Can you define it here and contrast it with Chisel type equivalence? Or is there another term "Chisel Type equivalence, including alignment check"?

Reply:
So right now, chisel type equivalence does not include an alignment check. In the future, I'd like to check alignment but that can only happen after we do the fixes to directionality I want to do after the connectable stuff. The way I'd categorize it is there is (1) a structural check (current type equivalence) and (2) a structural check + alignment check (future type equivalence). However, we do need both concepts regardless, so I'll clarify.